crates/hir-ty/src/diagnostics/expr.rs

```diff
@@ -235,6 +243,36 @@ impl ExprValidator {
         }
         pattern
     }
+
+    fn validate_block(&mut self, db: &dyn HirDatabase, expr: &Expr) {
+        let Expr::Block { statements, .. } = expr else { return };
+        let body = db.body(self.owner);
```

Comment:
We should move this into `ExprValidator` as a field and populate it early once. It's a minor perf nitpick but as this is a query every call of it with trigger an RwLock.